### PR TITLE
#1253 Added routes and unit tests for CommunicationItem

### PR DESCRIPTION
**Pull request by kalbfled**
# Description

These changes:
- Complete the CRUD implementation for the CommunicationItem model
- Include a migration to force uniqueness for two CommunicationItem columns
- Update the Postman collection to include all CommunicationItem routes
- Removes redundant lines in .talismanrc

I initially planned to make database queries using the newer SQLAlchemy v2.0 query syntax.  (We use version 1.4, which supports the new and legacy syntax.)  However, I came to understand that the syntax used throughout the app (i.e. ModelName.query.get(<id>), etc.) is actually provided by [flask-sqlalchemy](https://flask-sqlalchemy.palletsprojects.com/en/3.0.x/quickstart/#query-the-data), which [supports SQLAlchemy >= 1.4.18](https://flask-sqlalchemy.palletsprojects.com/en/3.0.x/changes/#version-3-0-0).

#1253

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] New unit tests pass
- [x] Manually tested in local environment with Postman
- [x] [Deployed](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/5213957332)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by ldraney — DISMISSED** [review]
I read through and understand, resolves the error, code looks good and passes tests; so despite being a lot I feel comfortable to approve!

File: app/communication_item/rest.py [hunk 1]
```diff
@@ -1,19 +1,180 @@
...
+
+    if communication_item is None:
+        return {
+            "errors": [
+                {
+                    "error": "NotFound",
+                    "message": f"Communication item {communication_item_id} does not exist.",
+                }
+            ]
+        }, 404
```
[the remark this answers is not in the document]
Comment (kalbfled) [reply app/communication_item/rest.py:92]: We do this all over the code base.  If we are going to stop doing that, I suggest we create a new ticket.

File: app/communication_item/rest.py [hunk 2]
```diff
@@ -1,19 +1,180 @@
...
```
[the remark this answers is not in the document]
Comment (kalbfled) [reply app/communication_item/rest.py:127]: ditto

File: app/communication_item/rest.py [hunk 3]
```diff
@@ -1,19 +1,180 @@
...
+        return {}, 202
+
+    return {
+        "errors": [
+            {
+                "error": "NotFound",
+                "message": f"Communication item {communication_item_id} was not deleted.",
+            }
+        ]
+    }, 404
```
[the remark this answers is not in the document]
Comment (kalbfled) [reply app/communication_item/rest.py:180]: ditto

File: app/communication_item/rest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cris-oddball) [reply app/communication_item/rest.py:92]: Maybe that is the 'api-key' routes return this instead:
```
{
  "message": "No result found",
  "result": "error"
}
```

File: app/communication_item/rest.py [hunk 4]
```diff
@@ -1,19 +1,180 @@
...
+        return {}, 202
+
+    return {
+        "errors": [
+            {
+                "error": "NotFound",
+                "message": f"Communication item {communication_item_id} does not exist.",
+            }
+        ]
+    }, 404
```
[the remark this answers is not in the document]
Comment (cris-oddball) [reply app/communication_item/rest.py:180]: see above

File: app/communication_item/rest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kalbfled) [reply app/communication_item/rest.py]: I changed the error message.

File: app/communication_item/rest.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kalbfled) [reply app/communication_item/rest.py]: I changed the error message.

File: app/communication_item/rest.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (kalbfled) [reply app/communication_item/rest.py]: I changed the error message.

**Review by cris-oddball — APPROVED** [review]

**Pull request merged**